### PR TITLE
Check if the diagram is valid before saving

### LINKED ISSUE #507: Validate diagrams before saving
#### Describe your issue
Currently the validation of a diagram only happens, if you evaluate the `enableSaveButton` and `disableSaveButton` Events, thrown by the Event aggregator. 

Regardless of the diagrams state, you can simply throw a `saveDiagram` message over the EventAggregator and the diagram will be saved. There is no dedicated check, if the diagram is valid. This behavior may leads to an issue like #501. 

In my opinion it would be better to add an additional check, if the diagram is valid, to the `saveDiagram` Method to prevent saving of an invalid diagram. 

#### Possible solution
Validate the diagram in the `saveDiagram` method before you actually save the diagram.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by robpal94**
## What did you change?
This PR adds a simple validation before the actual saving of the diagram in the `saveDiagram()` Method of the _processdef-detail_ module.

Related Feature: #502
Fixes #507

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/modules/processdef-detail/processdef-detail.ts [hunk 1]
```diff
@@ -200,6 +198,10 @@ export class ProcessDefDetail {
 
     this._validateXML();
 
+    if (this._diagramIsInvalid) {
+      return;
```
Comment (Paulomart) [inline src/modules/processdef-detail/processdef-detail.ts]: Why not throw a error here?

Comment (robpal94) [reply src/modules/processdef-detail/processdef-detail.ts]: Is the error handled somewhere? If I reject the promise here, I only get a stack trace in my browsers console so I thought, that an error from the `saveDiagram` Method is not properly handled yet.

Comment (Paulomart) [reply src/modules/processdef-detail/processdef-detail.ts]: Yeah, then we should handle the error somewhere where we call the method.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 2]
```diff
@@ -307,8 +307,8 @@ export class ProcessDefDetail {
   // TODO: Add Documentation.
   private async _saveDiagram(): Promise<void> {
 
-    // TODO: This needs to be refactored; _validateXML() does not seem to work properly.
-    this._validateXML();
+    // TODO: This needs to be refactored; _dropInvalidFormData() does not seem to work properly.
+    this._dropInvalidFormData();
```
Comment (robpal94) [inline src/modules/processdef-detail/processdef-detail.ts]: Is it really necessary to call it then? Im a bit afraid of calling a method, that is marked with _...does not work..._.

File: src/modules/processdef-detail/processdef-detail.ts [hunk 3]
```diff
@@ -304,10 +304,19 @@ export class ProcessDefDetail {
...
   }
 
-  // TODO: Add Documentation.
+  /**
+   * This method will save the diagram by using the ProcessEngineService.
+   *
+   * The user will be notified, about the outcome of the operation. Errors will be
+   * reported reasonably and a success message will be emitted.
+   *
+   * Saving is not possible, if _diagramIsInvalid has been set tu true.
```
Comment (robpal94) [inline src/modules/processdef-detail/processdef-detail.ts]: Typo: _set **to** true_

**Pull request merged**